### DISCUSSION TITLE
Unimarc analysis improvement: French binding type.

### DISCUSSION THREAD

**Discussion by meteos77** [textbrowser/biblioteq#216] [Ideas]
I submit a small modification in the biblioteq_marc.cc file

[biblioteq_marc.cc.zip](https://github.com/textbrowser/biblioteq/files/8672380/biblioteq_marc.cc.zip)

if I replace Relié by Hardcover and Broché by Paperback  it doesn't work anymore ?

modif line 129 - 135 and line 1069 - 1075

**Comment by textbrowser** [comment]
I uploaded the changes. Text which should be translated cannot be already translated. Let me know if it works.

**Comment by meteos77** [comment]
I'm focusing on SRU for the moment : MARC label data recovered by **SRU Bnf**

ISBN : 2-7096-2418-4 : the br. works
<mxc:datafield tag="020" ind1=" " ind2=" ">
      <mxc:subfield code="a">2-7096-2418-4</mxc:subfield>      
      <mxc:subfield code="b">br.</mxc:subfield>      
      <mxc:subfield code="d">22 EUR</mxc:subfield>   
</mxc:datafield>

ISBN : 2-7320-3771-0
 <mxc:datafield tag="020" ind1=" " ind2=" ">
       <mxc:subfield code="a">2-7320-3771-0</mxc:subfield>
       <mxc:subfield code="b">rel.</mxc:subfield>
       <mxc:subfield code="d">12 EUR</mxc:subfield>
</mxc:datafield>

ISBN : 2-7234-2388-3
<mxc:datafield tag="020" ind1=" " ind2=" ">
      <mxc:subfield code="a">2-7234-2388-3</mxc:subfield>
      <mxc:subfield code="b">cart.</mxc:subfield>
      <mxc:subfield code="d">8.99 EUR</mxc:subfield>
</mxc:datafield>  


- the information is in the **zone 020**  and  the subfield is well **br.** or **rel.**  a new type hardback with **cart.**

it doesn't work and I wonder why?
[Test-binding_type.sqlite.zip](https://github.com/textbrowser/biblioteq/files/8688376/Test-binding_type.sqlite.zip)

**Comment by textbrowser** [comment]
https://www.oclc.org/bibformats/en/0xx/020.html
https://www.loc.gov/marc/bibliographic/bd020.html

Codes b and d are not defined.

I will not be addressing every unique library.

**Reply by meteos77** [reply]
explanation of b and d  (Unimarc zone 10 to Marc21 -> zone 20)

![Unimarc_zone10-to-Marc21_zone20](https://user-images.githubusercontent.com/56431039/168343364-45530d40-cef5-413d-a9c3-8ff723d2ddf1.png)

**Reply by meteos77** [reply]
they are completely crazy :-)

**Reply by meteos77** [reply]
hats off to you for handling all this weirdness

**Reply by textbrowser** [reply]
Actually, the smart thing to do is: you have the physical item in your hands which suggests that you ought to know how to select incorrect properties after a query. The query result is a suggestion. That's the smart solution, not an endless pursuit of tweaking the software for individual results. However, that's me being rational.

**Reply by meteos77** [reply]
you assume that the book is in the hands.
we receive 500 books every 6 months, do you think that each book is examined to fill in or correct possible errors?
What is not automatically retrieved from the unimarc files is not usable with such a large amount of data.

**Reply by meteos77** [reply]
BiblioteQ provides fields, and if it can retrieve information from a library base via z3950 or SRU is that the purpose of these features?

on one side we have a filled database (for example Congress or Bnf), on the other side we have the BiblioteQ database. it's just a transfer of information, and I don't see what the human treatment brings between the 2.
double data entry, data entry error, speed, all lean towards a computer processing

**Reply by meteos77** [reply]
I recognize by this, the fantastic work you do to manage the 2 data formats

**Reply by meteos77** [reply]
to answer the subject of individual adjustment

French is spoken by 77 million people, so if BiblioteQ is multi-language, small adjustments like my modification on binding type can make it really well adapted to French.
https://en.wikipedia.org/wiki/List_of_languages_by_total_number_of_speakers?oldid=899012693

9 lines of code is the functionality binding type is 5 stars, it's beautiful

**Reply by textbrowser** [reply]
The first rule of data is never trust the source.

**Comment by textbrowser** [comment]
For 2-7234-2388-3, hardback (cart.) is produced.

**Reply by textbrowser** [reply]
For 2-7320-3771-0, hardback (rel.) is produced.

**Comment by textbrowser** [comment]
So working as intended.

**Reply by meteos77** [reply]
so it comes from my house
I have some checking to do, sorry for the inconvenience then

**Reply by meteos77** [reply]
for Codes b and d are not defined. in Unimarc, they are present on ifla.org 
page 3 : https://archive.ifla.org/VI/8/unimarc-concise-bibliographic-format-2008.pdf

**Comment by textbrowser** [comment]
Did you pull the latest source and remake the program?

**Reply by meteos77** [reply]
git clone + qmake + make
mysterious problems not improved with new compilation

I'm following a track concerning the link between database enumeration and interface language

**Reply by meteos77** [reply]
if I create paperback and hardcover in the database enumerations, it works fine with the english interface.

if I redo the SRU query with the french interface, it doesn't show me anything

if I put the interface in English, the 3 books are noted : livre broché in the interface of book details but in the table the value is still INCONNU

**Reply by meteos77** [reply]
ok it's solved

**Reply by meteos77** [reply]
explanation : the translation in French did not correspond to my statements in the database enumeration

the translations were
livre de poche et couverture rigide

I will switch them to livre relié et livre broché

In English the words are equivalent not in French and at the time of the translation, when you don't know exactly what is behind it, it is more difficult to put the right term.
